tree/dataframe/inc/ROOT/RDF/GraphUtils.hxx

```diff
@@ -35,20 +35,7 @@ class RRangeBase;
 namespace Internal {
 namespace RDF {
 class RColumnRegister;
-
 namespace GraphDrawing {
-std::shared_ptr<GraphNode>
-CreateDefineNode(const std::string &columnName, const ROOT::Detail::RDF::RDefineBase *columnPtr);
-
-std::shared_ptr<GraphNode> CreateFilterNode(const ROOT::Detail::RDF::RFilterBase *filterPtr);
-
-std::shared_ptr<GraphNode> CreateRangeNode(const ROOT::Detail::RDF::RRangeBase *rangePtr);
```

Comment:
`CreateDefineNode` was previously needed because of the friend on line 72 in the same file.

Currently `CreateDefineNode` is only defined (once) and only called (once) in `dataframe/src/RDFGraphUtils.cxx`.

It is the similar case for the other Creates as well.

Reply:
Right, `CreateDefineNode` is only needed in `RDFGraphUtils.cxx`. Then it needs to go in an anonymous namespace and be made static, to mark it as something that we only expect to call from that same translation unit.

`CreateFilterNode` and `CreateRangeNode` are called from `RFilter.hxx` and `RRange.hxx` respectively, if I read correctly.

Reply:
For `CreateFilterNode`: It is declared in `RFilter.hxx` and there is some method (GetGraph) that is calling `RFilter` in  `RFilter.hxx`.
Note that `RFilter` is already declared in `namespace GraphDrawing`.

The actual implementation of `RFilter.hxx` is in `RDFGraphUtils.cxx`, which is again in the same namespace and implements the function as declared in `RFilter.hxx`.

The case of `CreateRangeNode` is very similar.

I am not seeing the need of the declarations inside `GraphUtils.hxx`.

Moreover,  I just recompiled with "wrong declarations" in `GraphUtils.hxx`, i.e. not including the `fVisitedMap` as argument (other declarations have it); and all tests are passing.

Reply:
> I am not seeing the need of the declarations inside GraphUtils.hxx.

`CreateFilterNode` and `CreateRangeNode` are functions defined in this section of the source code that are used by other sections of the source code. As such, they need a declaration in the corresponding header. It's good practice and anyway it's how the rest of our code is written.

The fact that `RFilter.hxx` and `RRange.hxx` prefer to forward-declare the functions rather than include the corresponding header does not change that.

The fact that wrong declarations in `GraphUtils.hxx` do not lead to a compilation error points to a separate problem: definitions of free functions in the `.cxx` file should have the form:

```cpp
std::shared_ptr<GraphNode> GraphDrawing::CreateDefineNode
```
rather than
```cpp
namespace GraphDrawing {
std::shared_ptr<GraphNode> CreateDefineNode
}
```

(but fixing this is out of the scope of this PR)